src/prefect/core/task.py

```diff
@@ -764,6 +779,40 @@ def serialize(self) -> Dict[str, Any]:
         """
         return prefect.serialization.task.TaskSchema().dump(self)
 
+    # Accessors  ----------------------------------------------------------------
+
+    def get_item(self, item: Any, default: Any = NoDefault()) -> "Task":
+        """
+        Produces a Task that evaluates `self[key]` or returns the default value
+
+        Args:
+            - item (object): the object to use as an index for this task. It will be converted
+                to a Task if it isn't one already.
+            - default (object): the object to use as the default.
+
+        Returns:
+            - Task
+        """
+        args = (self, item) if isinstance(default, NoDefault) else (self, item, default)
+
+        return prefect.tasks.core.operators.GetItem().bind(*args)
+
+    def get_attr(self, item: str, default: Any = NoDefault()) -> "Task":
+        """
+        Produces a Task that evaluates `self.attr` or returns the default value
+
+        Args:
+            - item (str): the (possibly nested) attribute to retrieve as `self.attr`.
+                Nested attributes should be accessed via `.`-delimited strings.
+            - default (object): the object to use as the default.
+
+        Returns:
+            - Task
+        """
+        args = (self, item) if isinstance(default, NoDefault) else (self, item, default)
+
+        return prefect.tasks.core.operators.GetAttr().bind(*args)
```

Comment:
We already support `__getitem__` on a task for creating `GetItem` tasks. In a different issue automatic support for `GetAttr` was discussed but shot down, as typos in task attributes would silently create new tasks rather than erroring.

I don't think we want to add these accessor methods here, as I don't see this as a common pattern (prefect tasks generally aren't worth it for something as quick as a getitem/getattr).

Reply:
Understandable, although the proposed implementation doesn't use magic methods at all, rather a clearly defined api. The only reason I've added these two accessors was for convenience when wanting to define defaults if the given key doesn't exist. Although I get that tasks aren't worth creating for attribute access, but also sharing and ser/de large objects isn't desirable either when reusing the intermediate result in various parts of the workflow.

I can remove these and move them to my BaseTask that wraps Prefect's Task.